Update validators.md (#1635)
Shopify/quilt: packages/react-form-state/docs/validators.md

We can pass these directly into the `validators` prop of `<FormState />`.

> To validate the form automatically on form submit set the `validateOnSubmit` prop of `FormState` to be `true`.

Reviewer: ```suggestion
To use validators when submitting your form, set the `validateOnSubmit` prop of `<FormState />` to `true`. Here's an example:
```

I would also remove the line break to make a paragraph out of the sentence above, and then edit the example below to use this property.

        description: 'Cool product',
        quantity: 0,
      }}

>       validateOnSubmit

Reviewer: ```suggestion
      validateOnSubmit={true}
```
or is just specifying the property magic syntax for the same thing?

Author: Yup it is for the same thing, just some syntactical sugar.
